File: app/src/main/java/stock/com/ui/pojo/LobbyContestPojo.java
package stock.com.ui.pojo;

import android.os.Parcel;
import android.os.Parcelable;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class LobbyContestPojo extends BasePojo {
    @SerializedName("contest")
    @Expose
    public ArrayList<Contest> contest = null;

    public class Contest implements Serializable {

        @SerializedName("schedule_id")
        @Expose
        public int scheduleId; 
        @SerializedName("contest_teamremaining")
        @Expose
        public int contest_teamremaining;
        @SerializedName("contest_id")
        @Expose
        public int contestId;
        @SerializedName("exchangeid")
        @Expose
        public String exchangeid;
        @SerializedName("exchangename")
        @Expose
        public String exchangename;
        @SerializedName("exchangeimage")
        @Expose
        public String exchangeimage;
        @SerializedName("scheduleid")
        @Expose
        public int scheduleid;
        @SerializedName("confirm_winning")
        @Expose
        public String confirm_winning = "";
        @SerializedName("join_multiple")
        @Expose
        public String join_multiple= "";
        @SerializedName("schedule_name")
        @Expose
        public String scheduleName;
        @SerializedName("schedule_start")
        @Expose
        public String scheduleStart;
        @SerializedName("schedule_end")
        @Expose
        public String scheduleEnd;
        @SerializedName("winning_amount")
        @Expose
        public String winningAmount;
        @SerializedName("mid")
        @Expose
        public int  mid=0;
        @SerializedName("marketname")
        @Expose
        public String  marketname;

        public String winningAmount_temp="";

        public String getWinningAmount_temp() {
            return winningAmount_temp;
        }

        public void setWinningAmount_temp(String winningAmount_temp) {
            this.winningAmount_temp = winningAmount_temp;
        }

        @SerializedName("contestid")
        @Expose
        public int contestid;
        @SerializedName("category_id")
        @Expose
        public int categoryId;
        @SerializedName("contest_size")
        @Expose
        public int contestSize;
        @SerializedName("contest_type")
        @Expose
        public String contestType;
        @SerializedName("entry_fees")
        @Expose
        public String entryFees;
        @SerializedName("fees")
        @Expose
        public String fees;
        @SerializedName("win_amount")
        @Expose
        public String win_amount;
        @SerializedName("catname")
        @Expose
        public String catname;
        @SerializedName("rank")
        @Expose
        public String rank;
        @SerializedName("description")
        @Expose
        public String description;
        @SerializedName("teams_joined")
        @Expose
        public int teamsJoined;

        public int calculatePosition;
        public int date;
        @SerializedName("price_break")
        @Expose
        public ArrayList<PriceBreak> priceBreak = null;

        @SerializedName("total_winners")
        @Expose
        public String totalWinners;

        public void setCalculatePosition(int calculatePosition) {

            this.calculatePosition = calculatePosition;
        }

        public int getCalculatePosition() {
            return calculatePosition;
        }

        public void setDate(int date) {
            this.date = date;
        }

        public int getDate() {
            return date;
        }

        protected Contest(Parcel in) {
            if (in.readByte() == 0) {
                scheduleId = 0;
            } else {
                scheduleId = in.readInt();
            }
            if (in.readByte() == 0) {
                contestId = 0;
            } else {
                contestId = in.readInt();
            }
            if (in.readByte() == 0) {
                exchangeid = null;
            } else {
                exchangeid = in.readString();
            }
            exchangename = in.readString();
            exchangeimage = in.readString();
            if (in.readByte() == 0) {
                scheduleid = 0;
            } else {
                scheduleid = in.readInt();
            }
            scheduleName = in.readString();
            scheduleStart = in.readString();
            winningAmount = in.readString();
            if (in.readByte() == 0) {
                contestid = 0;
            } else {
                contestid = in.readInt();
            }
            if (in.readByte() == 0) {
                categoryId = 0;
            } else {
                categoryId = in.readInt();
            }
            if (in.readByte() == 0) {
                contestSize = 0;
            } else {
                contestSize = in.readInt();
            }
            contestType = in.readString();
            entryFees = in.readString();
            catname = in.readString();
            description = in.readString();
            if (in.readByte() == 0) {
                teamsJoined = 0;
            } else {
                teamsJoined = in.readInt();
            }
            if (in.readByte() == 0) {
                totalWinners ="";
            } else {
                totalWinners = in.readString();
            }
        }
    }


}

File: app/src/main/java/stock/com/ui/pojo/LobbyContestPojo.kt
package stock.com.ui.pojo

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


class LobbyContestPojo : BasePojo() {
    @SerializedName("contest")
    @Expose
    var contest: List<Contest>? = null

    inner class Contest {

        @SerializedName("schedule_id")
        @Expose
        var scheduleId: String? = null
        @SerializedName("contest_id")
        @Expose
        var contestId: String? = null
        @SerializedName("exchangeid")
        @Expose
        var exchangeid: String? = null
        @SerializedName("exchangename")
        @Expose
        var exchangename: String? = null
        @SerializedName("exchangeimage")
        @Expose
        var exchangeimage: String = ""
        @SerializedName("scheduleid")
        @Expose
        var scheduleid: String? = null
        @SerializedName("schedule_name")
        @Expose
        var scheduleName: String? = null
        @SerializedName("schedule_start")
        @Expose
        var scheduleStart: String = ""
        @SerializedName("winning_amount")
        @Expose
        var winningAmount: String? = null
        @SerializedName("contestid")
        @Expose
        var contestid: String? = null
        @SerializedName("category_id")
        @Expose
        var categoryId: String? = null
        @SerializedName("contest_size")
        @Expose
        var contestSize: String= ""
        @SerializedName("contest_type")
        @Expose
        var contestType: String? = null
        @SerializedName("entry_fees")
        @Expose
        var entryFees: String? = null
        @SerializedName("catname")
        @Expose
        var catname: String? = null
        @SerializedName("description")
        @Expose
        var description: String? = null
        @SerializedName("teams_joined")
        @Expose
        var teamsJoined: String = ""
        @SerializedName("price_break")
        @Expose
        var priceBreak: List<PriceBreak>? = null
        @SerializedName("total_winners")
        @Expose
        var totalWinners: String? = null

    }

    inner class PriceBreak {

        @SerializedName("contest_id")
        @Expose
        var contestId: String? = null
        @SerializedName("start_num")
        @Expose
        var startNum: String? = null
        @SerializedName("end_num")
        @Expose
        var endNum: String? = null
        @SerializedName("price_each")
        @Expose
        var priceEach: String? = null

    }
}